altair/utils/schemapi.py

```diff
@@ -1230,6 +1190,56 @@ def __dir__(self) -> list[str]:
         return sorted(chain(super().__dir__(), self._kwds))
 
 
+def _get_optional_modules(**modules: str) -> dict[str, ModuleType | None]:
+    """
+    Returns packages only if they have already been imported - otherwise they return `None`.
+
+    This is useful for `isinstance` checks.
+
+    For example, if `pandas` has not been imported, then an object is
+    definitely not a `pandas.Timestamp`.
+
+    Parameters
+    ----------
+    **modules
+        Keyword-only binding from `{alias: module_name}`.
+
+    Examples
+    --------
+    ::
+
+        ResultType = dict[Literal["pd", "pl"], ModuleType | None]
+        r: ResultType = _get_optional_modules(pd="pandas", pl="polars")
```

Comment:
> Minor: Should these keys also have the `_opt` suffix like this to follow the convention we used for pandas and numpy so far?

The `_opt` suffixes are simply encoding information that is now in the function name.

I think they were added to avoid confusion over reusing `(pd|np)`, which could now return `None`:
- https://github.com/vega/altair/blob/86d6b89d2071a5e0a6be919d235b1252bc472cf8/tools/schemapi/schemapi.py
- https://github.com/vega/altair/blob/2036b9940234cff7b7859fba4c139d0bf8dc5027/tools/schemapi/schemapi.py

I don't think this be a good fit for an example, since we're at the call site of the name binding.

This did make me re-evaluate the example I'd written though, and I think result is more useful in [`481790c` (#3530)](https://github.com/vega/altair/pull/3530/commits/481790c195adcd7b9141db032c30b7919c8f9d8e)